File: src/main/java/com/matag/game/turn/action/leave/PutIntoHandService.java
package com.matag.game.turn.action.leave;

import org.springframework.stereotype.Component;

import com.matag.game.cardinstance.CardInstance;
import com.matag.game.status.GameStatus;

@Component
public class PutIntoHandService {
  public void returnToHand(GameStatus gameStatus, CardInstance cardInstance) {
    var owner = gameStatus.getPlayerByName(cardInstance.getOwner());
    cardInstance.resetAllModifiers();
    owner.getHand().addCard(cardInstance);
  }
}

File: src/main/kotlin/com/matag/game/turn/action/leave/PutIntoHandService.kt
package com.matag.game.turn.action.leave

import com.matag.game.cardinstance.CardInstance
import com.matag.game.status.GameStatus
import org.springframework.stereotype.Component

@Component
class PutIntoHandService {
    fun returnToHand(gameStatus: GameStatus, cardInstance: CardInstance) {
        val owner = gameStatus.getPlayerByName(cardInstance.owner)
        cardInstance.resetAllModifiers()
        owner.hand.addCard(cardInstance)
    }
}
